packages/strapi-utils/lib/build-query.js

```diff
@@ -142,19 +147,28 @@ const normalizeClauses = (whereClauses, { model }) => {
  * @param {Object} options.rest - In case the database layer requires any other params pass them
  */
 const buildQuery = ({ model, filters = {}, ...rest }) => {
+  const { where, sort } = filters;
+
   // Validate query clauses
-  if (filters.where && Array.isArray(filters.where)) {
-    if (hasDeepFilters(filters.where, { minDepth: 2 })) {
+  if ([where, sort].some(Array.isArray)) {
+    if (hasDeepFilters({ where, sort, minDepth: 2 })) {
       strapi.log.warn(
         'Deep filtering queries should be used carefully (e.g Can cause performance issues).\nWhen possible build custom routes which will in most case be more optimised.'
       );
     }
 
-    // cast where clauses to match the inner types
-    filters.where = normalizeClauses(filters.where, { model });
+    if (sort) {
+      // Check that every field from the sort clauses is valid, throw with error otherwise
+      sort.forEach(({ field }) => getAssociationFromFieldKey({ model, field }));
```

Comment:
you can normalize the sorted fields here so the id becomes _id in mongo

Reply:
Thanks, done